### PR TITLE
LPD-25887 The picklist object field value can be an empty string. In this case, we should not return the picklist default value

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://liferay.atlassian.net/browse/LPD-25887

### CODE REVIEW

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -1428,6 +1431,83 @@ public void testAddObjectEntryWithAttachmentObjectField() throws Exception {
...
+				TestPropsValues.getUserId(), objectField.getObjectFieldId(),
+				ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+				ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
+
+			Assert.assertEquals(
+				StringPool.BLANK,
+				_getListEntryKey(
+					_defaultObjectEntryManager.getObjectEntry(
+						_simpleDTOConverterContext, objectDefinition,
+						objectEntryId)));
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1494]: I know that the ticket is related to GET, but we could also test the case in which we try to update an object entry (empty value for picklist) to verify if the default value will not be set as value("") and with a null value.
What do you think?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -1428,6 +1431,83 @@ public void testAddObjectEntryWithAttachmentObjectField() throws Exception {
 			objectDefinition.getObjectDefinitionId());
 	}
 
+	@Test
+	public void testAddObjectEntryWithPicklistObjectField() throws Exception {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1435]: Could you rename it to testAddOrUpdate... since we are adding more scenarios?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -4022,6 +4102,39 @@ private void _assertObjectEntriesSize2(long size) throws Exception {
 			objectEntries.toString(), size, objectEntries.size());
 	}
 
+	private long _assertObjectEntryProperties(
+			String actualPicklistObjectFieldValue,
+			String expectedPicklistObjectFieldValue,
+			ObjectDefinition objectDefinition)
+		throws Exception {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:4109]: What about renaming this method to:
- _addAndAssertObjectEntryWithPicklistObjectField()

What do you think?

I found 1 example of a method named searchAndAssertLength() in a test class :eyes:

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 4]
```diff
@@ -1428,6 +1431,101 @@ public void testAddObjectEntryWithAttachmentObjectField() throws Exception {
...
+
+			Assert.assertEquals(
+				StringPool.BLANK,
+				_getListEntryKey(
+					_defaultObjectEntryManager.getObjectEntry(
+						_simpleDTOConverterContext, objectDefinition,
+						objectEntryId2)));
+
+			_updateAndAssertObjectEntryWithPicklistObjectField(
+				null, null, objectDefinition, objectEntryId1);
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1503]: I thought that this case would return the default value :/

**Review by marcelabc — APPROVED** [review]
Overall, LGTM!
I just have 1 question

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 4 above)
Comment (carolmariaabb) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1503]: we only add the default value when creating a new entry

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward:force

**Comment by carolmariaabb** [comment]
ci:forward:force
